src/topology/mod.rs

```diff
@@ -228,32 +248,40 @@ impl RunningTopology {
             to_remove_change_add(&self.config.sources, &new_config.sources);
 
         // First pass to tell the sources to shut down.
+        let mut source_shutdown_complete_futures = Vec::new();
+        // TODO: Once all Sources properly look for the ShutdownSignal, up this time limit to something
```

Comment:
Lets make sure we have issues for some of these TODOs

Reply:
Created https://github.com/timberio/vector/issues/2108 and https://github.com/timberio/vector/issues/2109